EESSI-2023.06-extend.eb

```diff
@@ -0,0 +1,85 @@
+easyblock = 'Binary'
+
+name = 'EESSI'
+version = '2023.06'
+versionsuffix = '-extend'
+
+homepage = 'https://eessi.github.io/docs/'
+
+description = """
+ The goal of the European Environment for Scientific Software Installations
+ (EESSI, pronounced as "easy") is to build a common stack of scientific
+ software installations for HPC systems and beyond, including laptops,
+ personal workstations and cloud infrastructure.
+
+ This module allows you to extend EESSI using the same configuration for
+ EasyBuild as EESSI itself uses. It installs the modules in a special location
+ that is configurable in the CVMFS configuration (so can be on a shared
+ filesystem), and automatically found when using the default EESSI environment. 
+"""
+
+toolchain = SYSTEM
+
+source_urls = None
+sources = ['eb_hooks.py']
+checksums = ['8ae609f99b6953beae89aa1945913686c86c156509dbc55e2b6b8017b13fcf66']
+
+# Don't extend the PATH
+prepend_to_path = None
+
+# All the dependencies we filter in EESSI
+local_deps_to_filter = "Autoconf,Automake,Autotools,binutils,bzip2,DBus,flex,gettext,gperf,help2man,intltool,libreadline,libtool,Lua,M4,makeinfo,ncurses,util-linux,XZ,zlib"
+local_arch_specific_deps_to_filter = {'aarch64': ',yasm', 'x86_64': ''}
+local_deps_to_filter += local_arch_specific_deps_to_filter[ARCH]
+
+modextravars = {
+    'EASYBUILD_FILTER_DEPS': local_deps_to_filter,
+    'EASYBUILD_IGNORE_OSDEPS': '1',
+    'EASYBUILD_DEBUG': '1',
+    'EASYBUILD_TRACE': '1',
+    'EASYBUILD_ZIP_LOGS': 'bzip2',
+    'EASYBUILD_RPATH': '1',
+    'EASYBUILD_FILTER_ENV_VARS': 'LD_LIBRARY_PATH',
+    'EASYBUILD_READ_ONLY_INSTALLDIR': '1',
+    'EASYBUILD_MODULE_EXTENSIONS': '1',
+    'EASYBUILD_EXPERIMENTAL': '1',
+    'EASYBUILD_HOOKS': '%(installdir)s/eb_hooks.py',
+}
+
+# Need a few other variables, but they are more dynamic
+# EASYBUILD_SYSROOT=${EPREFIX}
+# EASYBUILD_PREFIX=${WORKDIR}/easybuild
+# EASYBUILD_INSTALLPATH=${EESSI_PREFIX}/software/${EESSI_OS_TYPE}/${EESSI_SOFTWARE_SUBDIR}
```

Comment:
Default for `$EASYBUILD_INSTALLPATH` should be `$HOME/eessi`, unless `$EASYBUILD_INSTALLPATH` is already set.

Maybe something more fancy is needed for `host_injections` or installing in writable overlay (like the bot would)

Reply:
I've implemented something that covers 3 scenarios: installation into CVMFS, installation under `host_injections` and installation as a normal user